stock_account_product_run_fifo_hook/hooks.py

```diff
@@ -0,0 +1,348 @@
+        new_standard_price = 0
+        tmp_value = 0  # to accumulate the value taken on the candidates
+        taken_data = {}
+        for candidate in candidates:
+            qty_taken_on_candidate = min(
+                qty_to_take_on_candidates, candidate.remaining_qty
+            )
+            taken_data[candidate.id] = {"quantity": qty_taken_on_candidate}
+            candidate_unit_cost = candidate.remaining_value / candidate.remaining_qty
+            new_standard_price = candidate_unit_cost
+            value_taken_on_candidate = qty_taken_on_candidate * candidate_unit_cost
+            value_taken_on_candidate = candidate.currency_id.round(
+                value_taken_on_candidate
+            )
+            taken_data[candidate.id].update(
+                {
+                    "value": value_taken_on_candidate,
+                }
+            )
+            new_remaining_value = candidate.remaining_value - value_taken_on_candidate
+            candidate_vals = {
+                "remaining_qty": candidate.remaining_qty - qty_taken_on_candidate,
+                "remaining_value": new_remaining_value,
+            }
+            # Start Hook Prepare Candidate
+            candidate_vals = self._run_fifo_prepare_candidate_update(
+                candidate,
+                qty_taken_on_candidate,
+                value_taken_on_candidate,
+                candidate_vals,
+            )
+            # End Hook Prepare Candidate
+            candidate.write(candidate_vals)
+
+            qty_to_take_on_candidates -= qty_taken_on_candidate
+            tmp_value += value_taken_on_candidate
+            if float_is_zero(
+                qty_to_take_on_candidates, precision_rounding=self.uom_id.rounding
+            ):
+                if float_is_zero(
+                    candidate.remaining_qty, precision_rounding=self.uom_id.rounding
+                ):
+                    next_candidates = candidates.filtered(
+                        lambda svl: svl.remaining_qty > 0
+                    )
+                    new_standard_price = (
+                        next_candidates
+                        and next_candidates[0].unit_cost
+                        or new_standard_price
+                    )
+                break
+
+        # Update the standard price with the price of the last used candidate,
+        # if any.
+        # START HOOK update standard price
+        if hasattr(self, "_price_updateable"):
+            if self._price_updateable(new_standard_price):
+                self.sudo().with_company(company.id).with_context(
+                    disable_auto_svl=True
+                ).standard_price = new_standard_price
+        else:
+            if new_standard_price and self.cost_method == "fifo":
+                self.sudo().with_company(company.id).with_context(
+                    disable_auto_svl=True
+                ).standard_price = new_standard_price
+        # END HOOK update standard price
```

Comment:
```suggestion
        # Fifo out will change the AVCO value of the product. So in case of out,
        # we recompute it base on the remaining value and quantities.
        if self.cost_method == 'fifo' or (hasattr(self, "_price_updateable") and self._price_updateable(new_standard_price)):
            quantity_svl = sum(candidates.mapped('remaining_qty'))
            value_svl = sum(candidates.mapped('remaining_value'))
            product = self.sudo().with_company(company.id).with_context(disable_auto_svl=True)
            if float_compare(quantity_svl, 0.0, precision_rounding=self.uom_id.rounding) > 0:
                product.standard_price = value_svl / quantity_svl
            elif candidates and not float_is_zero(qty_to_take_on_candidates, precision_rounding=self.uom_id.rounding):
                product.standard_price = new_standard_price
```
due to https://github.com/odoo/odoo/commit/9cd316f038dc95051814ede9474f8f15f7d165df